### PR TITLE
WIP debugPrettyPrint

### PR DESCRIPTION
**Pull request by swoogles**
Super rough idea at this point, but I've often found myself pulling in lihayoi's pretty print library in projects during development, and it always sticks out like a sore thumb in my dependencies.

Just curious if some incarnation of this would be desirable.

### CODE REVIEW

File: core-tests/shared/src/test/scala/zio/ZIOSpec.scala [hunk 1]
```diff
@@ -4176,6 +4176,14 @@ object ZIOSpec extends ZIOBaseSpec {
...
         }
       }
+    ),
+    suite("debugPrettyPrint")(
+      test("experiments") (
+        for {
+          _ <- ZIO.succeed(List(List("Apple", "Bear", "Correspondence"), List(1, 2, 3))).debugPrettyPrint
+          _ <- ZIO.succeed(Map(("Bill", 1), ("Kit", 2), ("Adam", 3))).debugPrettyPrint
+        } yield assertCompletes
+      )
```
Comment (swoogles) [inline core-tests/shared/src/test/scala/zio/ZIOSpec.scala:4186]: This currently outputs:
```
List(List("Apple", "Bear", "Correspondence"), List(1, 2, 3))
Map(
  "Bill" -> 1,
  "Kit" -> 2,
  "Adam" -> 3
)
```
The `Map` is good, but I would also love to see nested `List`s split onto new lines too. Though that might be straying too far from the intent of this `PrettyPrint` implementation.

File: core/shared/src/main/scala/zio/ZIO.scala [hunk 2]
```diff
@@ -445,6 +445,11 @@ sealed trait ZIO[-R, +E, +A]
       .tap(value => ZIO.succeedNow(println(value)))
       .tapErrorCause(error => ZIO.succeedNow(println(s"<FAIL> $error")))
 
+  final def debugPrettyPrint(implicit trace: Trace): ZIO[R, E, A] =
+    self
+      .tap(value => ZIO.succeedNow(println(PrettyPrint(value))))
+      .tapErrorCause(error => ZIO.succeedNow(println(s"<FAIL> $error")))
```
Comment (swoogles) [inline core/shared/src/main/scala/zio/ZIO.scala:451]: Maybe we would want to `PrettyPrint` the failure case too

**Review by adamgfraser — COMMENTED** [review]
I like the idea of improving the rendering but I think if we are going to do this we should just do it for the existing `debug` family of operators rather than adding another set of variants.

**Comment by jdegoes** [comment]
Will close this for now in favor of doing this with `debug` as per @adamgfraser's suggestion. Thanks for your work on this!

**Pull request closed without merging**